packages/devtools/src/common/retry.ts

```diff
@@ -0,0 +1,104 @@
+import { createModuleLogger, printJson } from '@layerzerolabs/io-devtools'
+import assert from 'assert'
+import { backOff } from 'exponential-backoff'
+
+export type OnRetry<TInstance, TArgs extends unknown[] = unknown[]> = (
+    attempt: number,
+    numAttempts: number,
+    error: unknown,
+    target: TInstance,
+    method: string,
+    args: TArgs
+) => boolean | void | undefined
+
+export interface RetriableConfig<TInstance = unknown> {
+    /**
+     * Enable / disable the retry behavior
+     */
+    enabled?: boolean
+    /**
+     * The maximum delay, in milliseconds, between two consecutive attempts.
+     *
+     * @default Infinity
+     */
+    maxDelay?: number
+    /**
+     * Number of time the method call will be retried. The default is 3
+     *
+     * @default 3
+     */
+    numAttempts?: number
+    /**
+     * Callback called on every failed attempt.
```

Comment:
Can you document whether this happens right after a failure or right before a retry?

Reply:
I sure cannot, where would I get this information? Do you have a number I could call?

Reply:
What is this, the 90s? There is a chatbot that doesnt know either, but would love to help.